### PR TITLE
Rebrand 'pattern' as 'match pattern' throughout client codebase

### PR DESCRIPTION
**Pull request by StachuDotNet**
also, prepare for backend serializing Match Pattern cases as `MPString` rather than `PString`.

This replaces #4467. This is in preparation for #3402.

### CODE REVIEW

**Comment by StachuDotNet** [comment]
I tried my best to break this down into commits that could be read in-order. That ended up resulting in many commits.

There's a chance I missed something, but I'm fairly certain I've gotten 95%+ of the updates that make sense. Throughout this, I've reviewed all instances of `"pattern"` in our `client\` codebase.

**Comment by StachuDotNet** [comment]
- [ ] failing int. test: `fluid_ctrl_left_on_empty_match`

File: client/src/fluid/Fluid.res [hunk 1]
```diff
@@ -1175,130 +1174,132 @@ let caretTargetForStartOfExpr = (astPartId: id, ast: FluidAST.t): CT.t =>
...
- * on any tokenization functions. */
-let rec caretTargetForEndOfPattern = (pattern: fluidPattern): CT.t =>
-  switch pattern {
-  | PVariable(id, varName) => {
-      astRef: ARPattern(id, PPVariable),
+@ocaml.doc(
+  "returns a caretTarget representing caret placement at the very end
+  of the expression in `matchPattern`.
+
+  The concept of "
```
Comment (pbiggar) [inline client/src/fluid/Fluid.res:1201]: Weird formatting here

**Review by pbiggar — APPROVED** [review]
Nice! Please merge ASAP

File: client/src/fluid/Fluid.res [hunk 2]
```diff
@@ -1175,130 +1174,132 @@ let caretTargetForStartOfExpr = (astPartId: id, ast: FluidAST.t): CT.t =>
...
+  very
+  end
+  " is related to an understanding of the tokenization
```
Comment (StachuDotNet) [inline client/src/fluid/Fluid.res:1204]:
```suggestion
  The concept of 'very end' is related to an understanding of the tokenization
```

File: client/src/fluid/Fluid.res (hunk 1 above)
Comment (StachuDotNet) [reply client/src/fluid/Fluid.res]: Ah - that's what happens when you use `"`s in an `@ocaml.doc()`, it seems.

**Pull request merged**